src/ProposalTypesConfigurator.sol

```diff
@@ -21,7 +21,7 @@ contract ProposalTypesConfigurator is IProposalTypesConfigurator {
 
     mapping(uint8 proposalTypeId => ProposalType) internal _proposalTypes;
     mapping(uint8 proposalTypeId => bool) internal _proposalTypesExists;
-    mapping(uint8 proposalTypeId => Scope[]) public scopes;
+    mapping(uint8 proposalTypeId => mapping(bytes32 typeHash => Scope)) public scopes;
     mapping(uint8 proposalTypeId => mapping(bytes32 typeHash => bool)) public scopeExists;
```

Comment:
we are repeating `uint8 proposalTypeId => mapping(bytes32 typeHash` so we should combine both by having a struct that combines `Scope` and a boolean that indicates if it exists or not